File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP1-2/src/constructeurAuto/Constructeur.java
package constructeurAuto;

import Usines.Usine;

public class Constructeur {

	private static int id = 0;
	private int idConstructeur;
	private String nom;
	private Usine[] usines;
	private Stock[] stocks;
	
	public Constructeur(String nom, Usine usine, Stock stock) {
		this.idConstructeur = id;
		Constructeur.id ++;
		this.nom = nom;
		this.usines = new Usine[1];
		this.usines[0] = usine;
		this.stocks = new Stock[1];
		this.stocks[0] = stock;
	}
	
	public Constructeur(String nom, Usine[] usines, Stock stock) {
		this.idConstructeur = id;
		Constructeur.id ++;
		this.nom = nom;
		this.usines = usines;
		this.stocks = new Stock[1];
		this.stocks[0] = stock;
	}
	
	public Constructeur(String nom, Usine usine, Stock[] stocks) {
		this.idConstructeur = id;
		Constructeur.id ++;
		this.nom = nom;
		this.usines = new Usine[1];
		this.usines[0] = usine;
		this.stocks = stocks;
	}
	
	public Constructeur(String nom, Usine[] usines, Stock[] stocks) {
		this.idConstructeur = id;
		Constructeur.id ++;
		this.nom = nom;
		this.usines = usines;
		this.stocks = stocks;
	}
	
	public Constructeur(Constructeur constructeur) {
		this.idConstructeur = id;
		Constructeur.id ++;
		this.nom = constructeur.nom;
		this.usines = constructeur.usines;
		this.stocks = null;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public Usine[] getUsines() {
		return usines;
	}
	
	public void setUsines(Usine usine) {
		this.usines = new Usine[1];
		this.usines[0] = usine;
	}

	public void setUsines(Usine[] usines) {
		this.usines = usines;
	}

	public Stock[] getStocks() {
		return stocks;
	}
	
	public void setStocks(Stock stock) {
		this.stocks = new Stock[1];
		this.stocks[0] = stock;
	}

	public void setStocks(Stock[] stocks) {
		this.stocks = stocks;
	}

	public int getIdConstructeur() {
		return idConstructeur;
	}
	
	public String usinesString() {
		String stringListe = "";
		for(int i=0; i<this.usines.length; i++) {
			stringListe = this.usines[i].toString() + "\n";
		}
		return stringListe;
	}
	
	public String stocksString() {
		String stringListe = "";
		for(int i=0; i<this.stocks.length; i++) {
			stringListe = this.stocks[i].toString() + "\n";
		}
		return stringListe;
	}

	@Override
	public String toString() {
		return "Constructeur [idConstructeur=" + idConstructeur + ", nom=" + nom + ", usines=" + this.usinesString() + ", stocks=" + this.stocksString() + "]";
	}
	
}

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP2/ConstructeurAuto/Constructeur.php
<?php
class Constructeur{
    static $id = 0;
    protected $idConstructeur;
    protected $nom;
    protected $usines = array();
    protected $stocks = array();

    public function constructeur(String $nom = "constructeurAuto", array $usines = null, array $stocks = null){
        self::$id++;
        $this->idConstructeur = self::$id;
        $this->nom = $nom;
        if($usines != null){
           for($i = 0; $i < count($this->usines);$i++){
                array_push($this->usines, $usines[$i]);
            } 
        }
        if($stocks != null){
            for($i = 0; $i < count($this->stocks);$i++){
                array_push($this->stocks, $stocks[$i]);
            }
        }
    }

    public function getNom() : String {return $this->nom;}
    public function getUsines() : array {return $this->usines;}
    public function getStocks() : array {return $this->stocks;}
    public function getIdConstructeur() : int{return $this->idConstructeur;}

    public function setNom(String $nom = "constructeurAuto"){$this->nom = $nom;}
    public function setUsines(array $usines = null){$this->usines = $usines;}
    public function setStocks(array $stock = null){$this->stocks = $stock;}

//Méthodes
    public function __toString() : String {
        return "idConstructeur : " . $this->getIdConstructeur() . " nom : "  . $this->getNom(). " usines : " . $this->usinesString() . " stocks : " . $this->stocksString();
    }

    public function usinesString() : String {
        $stringListe = "";
        $nb = count($this->usines);
        if($nb != 0){
            for($i = 0; $i < $nb; $i++){
                $stringListe .= $this->usines[$i]->__toString();
            }
        }
        else{$stringListe .= "Aucune Usine reliée.";}
        return $stringListe;
    }

    public function stocksString() : String {
        $stringListe = "";
        $nb = count($this->stocks);
        if($nb != 0){
            for($i = 0; $i < $nb; $i++){
                $stringListe .= $this->stocks[$i]->__toString() . "\t";
            }
        }
        else{$stringListe .= "Aucun Stock relié.";}
        
        return $stringListe;
    }
}
?>